plugins/out_s3/s3.c

```diff
@@ -1488,24 +1519,28 @@ int get_md5_base64(char *buf, size_t buf_size, char *md5_str, size_t md5_str_siz
     return 0;
 }
 
-static struct multipart_upload *get_upload(struct flb_s3 *ctx,
-                                           const char *tag, int tag_len)
+static struct multipart_upload *get_upload(struct flb_s3 *ctx,const char *tag, int tag_len)
 {
     struct multipart_upload *m_upload = NULL;
     struct multipart_upload *tmp_upload = NULL;
     struct mk_list *tmp;
     struct mk_list *head;
+    char create_time[20];
+    struct tm now_time;
 
     mk_list_foreach_safe(head, tmp, &ctx->uploads) {
         tmp_upload = mk_list_entry(head, struct multipart_upload, _head);
-
+        /* Convert time_t to string */
+        strftime(create_time, 20, "%Y-%m-%d %H:%M:%S", localtime_r(&(tmp_upload->init_time), &now_time));
         if (tmp_upload->upload_state == MULTIPART_UPLOAD_STATE_COMPLETE_IN_PROGRESS) {
             continue;
         }
-        if (tmp_upload->upload_errors >= MAX_UPLOAD_ERRORS) {
+        if (tmp_upload->upload_errors >= ctx->ins->retry_limit) {
             tmp_upload->upload_state = MULTIPART_UPLOAD_STATE_COMPLETE_IN_PROGRESS;
-            flb_plg_error(ctx->ins, "Upload for %s has reached max upload errors",
-                          tmp_upload->s3_key);
+            flb_plg_error(ctx->ins,
```

Comment:
why is this an error and everywhere else it is a warning?

Reply:
Because before, here is an error.